cfme/networks/views.py

```diff
@@ -374,6 +374,23 @@ class my_filters(Accordion):  # noqa
         tree = ManageIQTree()
 
 
+class NetworkEntityWithNetworkPortView(NetworkPortView):
+    """ Represents a network entity with NetworkPorts page """
+
+    @property
+    def is_displayed(self):
+        obj = self.context['object']
+        from cfme.networks.subnet import Subnet
+        matched_title = ('{name} (All Network Ports)'.format(
+            name=self.context['object'].name) == self.entities.title.text
+        )
+        if isinstance(obj, Subnet):
+            matched_navigation = self.navigation.currently_selected == ['Networks', 'Subnets']
+        else:
+            matched_navigation = False
```

Comment:
As we are restricting by checking navigation tree pointing to `Subnets`. I think better to use `SubnetNetworkPortView` and remove this check of `Subnet` class. This view will not going to applicable to other `NetworkEntitys`.

Reply:
Sure, we meant to create a view which be applicable for other network entities. But sure, since is currently not used anywhere else.